src/components/layout/PrimaryLayout/PrimaryLayout.js

```diff
@@ -22,32 +24,38 @@ export function PrimaryLayout ({
   showSideMenu = true,
   showAlphaFlag = true
 }) {
+  const [isHposConnectionAlive, setIsHposConnectionAlive] = useState()
   const { setIsConnected, isConnected } = useConnectionContext()
   const { setCurrentUser } = useCurrentUserContext()
+  const { newMessage } = useFlashMessageContext()
+  const { push } = useHistory()
 
   const onError = ({ graphQLErrors }) => {
     const { isHposConnectionActive } = graphQLErrors
-    setIsConnected(isHposConnectionActive)
-  }
-
-  const onCompleted = ({ hposSettings }) => {
-    if (hposSettings) setIsConnected(true)
+    if (!isHposConnectionActive) {
+      setIsHposConnectionAlive(false)
+    } else {
+      setIsHposConnectionAlive(isHposConnectionActive)
+    }
   }
 
-  const { data: { hposSettings: settings = {} } = {} } = useQuery(HposSettingsQuery, { pollInterval: 10000, onCompleted, onError, notifyOnNetworkStatusChange: true, ssr: false })
-  const { newMessage } = useFlashMessageContext()
+  const { data: { hposSettings: settings = {} } = {} } = useQuery(HposSettingsQuery, { pollInterval: 10000, onError, notifyOnNetworkStatusChange: true, ssr: false })
 
   useEffect(() => {
-    if (!isConnected) {
+    setIsConnected(isHposConnectionAlive && wsConnection)
+    if (isConnected === false) {
```

Comment:
this is a little tricky, because you're setting `isConnected` and then immediately using the *old* value. In practice it's going to work here, but generally you should avoid this. Simplest way in this case would be by replacing

`if (isConnected === false) {`
with

`if (!(isHposConnectionAlive && wsConnection)) {`

Might be more readable to do

```
const newIsConnected = isHposConnectionAlive && wsConnection
setIsConnected(newIsConnected)
    if (!newIsConnected) {
```

Reply:
Now wondering about moving the setting of `isConnected` into a separate wrapper component, like `LoadCurrentUser`. We already have it in context, so that would be simple enough, and it takes the responsibility away from `PrimaryLayout`, which makes sense to me.

-- this comment is not blocking.